hash_practice/exercises.py

```diff
@@ -1,29 +1,81 @@
+import collections
 
+# def grouped_anagrams(strings):
 def grouped_anagrams(strings):
     """ This method will return an array of arrays.
         Each subarray will have strings which are anagrams of each other
         Time Complexity: ?
         Space Complexity: ?
     """
-    pass
+    # Time Complexity: O(nmlog(m)) where n is the length of the strings and m is the length of  the character in the string

```

Comment:
This is correct but we're also just accepting O(n). Yes, m has an effect on how long this takes, it's true, but since we're assuming English words, the size of m is rather limited (only 5 letters per average in a word). Comparatively, the size of n is unbounded (could be hundreds or thousands) so it is the more important factor in this algorithm so we can just treat it as O(n).